ui/src/helpers/get-run-duration.ts

```diff
@@ -36,8 +36,18 @@ export function calculateDuration(
   const minutes = Math.floor((durationSec % 3600) / 60);
   const seconds = durationSec % 60;
 
-  // Format the duration as "hours:minutes:seconds"
-  const formattedDuration = `${hours.toString().padStart(2, '0')}:${minutes.toString().padStart(2, '0')}:${seconds.toString().padStart(2, '0')}`;
+  // Format the duration as "xh ym zs", omitting zero values
+  let formattedDuration = '';
+  if (hours > 0) {
+    formattedDuration += `${hours}h `;
+  }
+  if (minutes > 0) {
```

Comment:
I just had a look how the Kotlin Duration class formats the string if some of the values are 0:

```
0h 0m 0s -> 0s
0h 0m 10s -> 10s
0h 10m 10s -> 10m 10s
0h 10m 0s -> 10m
10h 0m 0s -> 10h
10h 0m 10s -> 10h 0m 10s
10h 10m 0s -> 10h 10m
10h 10m 10s -> 10h 10m 10s
```

So it leaves out zero values at the start and the end but not in the middle which makes sense to me because `10h 10s` looks a bit odd. Adapting the conditions as below should align the behavior:

```ts
if (hourse > 0) ...
if (minutes > 0 || (hours > 0 && seconds > 0)) ...
if (seconds > 0 || (hours == 0 && minutes == 0)) ...
```

For reference I'll post the Kotlin code below, but it is more complex because it also handles days , fractional seconds and negative values:

```kotlin
                    val hasDays = days != 0L
                    val hasHours = hours != 0
                    val hasMinutes = minutes != 0
                    val hasSeconds = seconds != 0 || nanoseconds != 0
                    var components = 0
                    if (hasDays) {
                        append(days).append('d')
                        components++
                    }
                    if (hasHours || (hasDays && (hasMinutes || hasSeconds))) {
                        if (components++ > 0) append(' ')
                        append(hours).append('h')
                    }
                    if (hasMinutes || (hasSeconds && (hasHours || hasDays))) {
                        if (components++ > 0) append(' ')
                        append(minutes).append('m')
                    }
                    if (hasSeconds) {
                        if (components++ > 0) append(' ')
                        when {
                            seconds != 0 || hasDays || hasHours || hasMinutes ->
                                appendFractional(seconds, nanoseconds, 9, "s", isoZeroes = false)
                            nanoseconds >= 1_000_000 ->
                                appendFractional(nanoseconds / 1_000_000, nanoseconds % 1_000_000, 6, "ms", isoZeroes = false)
                            nanoseconds >= 1_000 ->
                                appendFractional(nanoseconds / 1_000, nanoseconds % 1_000, 3, "us", isoZeroes = false)
                            else ->
                                append(nanoseconds).append("ns")
                        }
                    }
                    if (isNegative && components > 1) insert(1, '(').append(')')
```